core-blocks/image/edit.js

```diff
@@ -179,31 +198,53 @@ class ImageEdit extends Component {
 		return get( this.props.image, [ 'media_details', 'sizes' ], {} );
 	}
 
+	switchToEditing() {
+		const { url } = this.props.attributes;
+
+		this.props.setAttributes( {
+			url: undefined,
+			alt: undefined,
+			id: undefined,
+			caption: undefined,
+			src: url,
+		} );
+	}
+
 	render() {
 		const { attributes, setAttributes, isLargeViewport, isSelected, className, maxWidth, noticeOperations, noticeUI, toggleSelection } = this.props;
 		const { url, alt, caption, align, id, href, width, height } = attributes;
+		const isExternal = url ? parse( wpApiSettings.root ).host !== parse( url ).host : true;
```

Comment:
We'd want to avoid access to globals as much as possible, or at least centralize their use. This looks a bit like a utility function which would belong in something like [the `@wordpress/url` package](https://github.com/WordPress/packages/tree/master/packages/url).

Do we want to ensure that the URL is specifically on a different host than the REST API base, or would it be enough to do something like `url.indexOf( window.location.origin ) === 0`?